.github/workflows/refresh-lock-and-linter-dependencies.yml

```diff
@@ -0,0 +1,94 @@
+# SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+name: Refresh lock and linter Dependencies
+
+on:
+  workflow_call:
+    inputs:
+      create-pr:
+        type: boolean
+        description: "Whether to create a pull request with the changes"
+        required: false
+        default: false
+  workflow_dispatch:
+    inputs:
+      create-pr:
+        type: boolean
+        description: "Whether to create a pull request with the changes"
+        required: false
+        default: false
+  schedule:
+    - cron: "0 2 * * 0" # Every Sunday at 2:00 UTC
+
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-linter-dependencies
+  cancel-in-progress: true
+
+
+jobs:
+  refresh-lock-and-linter-dependencies:
+    name: Refresh lock and linter Dependencies
+    runs-on: ubuntu-24.04
+    permissions:
+      contents: write
+      pull-requests: write
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v6
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v7
+
+      - name: upgrade uv lock
+        run: uv lock --upgrade  
+
+      - name: Install linters from uv
+        run: |
+          uv sync --only-group lint
+      
+      - name: retrieve versions
+        id: versions
+        run: |
+          grab_installed_version() {
+            local pkg="$1"
+            local version
+            version=$(uv pip freeze | grep -i "^$pkg==" | head -n1 | cut -d= -f3)
+            echo "${pkg}=$version" >> "$GITHUB_OUTPUT"
+          }
+          grab_installed_version "ruff"
+          grab_installed_version "clang-format"
+          grab_installed_version "mypy"
+          grab_installed_version "gersemi"
+
+      - name: Update .pre-commit-config.yaml
```

Comment:
I also just found this: https://pre-commit.com/#updating-hooks-automatically

But this PR is probably the better path going forward since otherwise the versions may diverge again.

One risk that this PR has is that the version on precommit may be lagging by a couple days, however, I don't think that's an issue because it means that the `uv` version should probably not be synced yet anyways.

Reply:
I can add a pre-commit install before creating the PR. If that fails, the refresh will fail. And it will wait until next time.

Reply:
That auto-update will not garantee the same version as in the dependency group of uv.